### DISCUSSION TITLE
React Clients

### DISCUSSION THREAD

**Discussion by lostcarpark** [LVerhulst4321/PlanZ#58] [General]
I'm looking at implementing participant links, and I think a React front end for them would be nice. This would be embedded in the "My Profile" page. Initially this would just display a panel for specifying links, but later it could be expanded to take over other parts of the profile page, and eventually handle the full participant profile.

I'm wondering whether I should add it into BC's volunteer React front end, or make it a standalone React app.

Advantages of adding to volunteer one:
- Keeps all React in one place.
- Only one React compilation required.
- Only one React component to load, so that once cached, switching between pages using React should be quicker.

Advantages of standalone app:
- Means each React app has tightly defined scope.
- If we move to a modular structure, React components can be separated more easily.
- Each React app should be smaller, so that components for pages that the user isn't visiting don't get loaded, making initial load quicker.

I have to admit I'm torn, but probably leaning towards making it standalone for easier modularisation in future. It would mean there would be multiple components to build, but I think this could be mitigated with a build script.

Any thoughts?

**Comment by bcholmes** [comment]
Two additional thoughts:

1. We'll have certain utility components that'll probably cross functional areas; where would those live and how would we manage dependencies?
2. I confess that I am concerned about overwhelming people's ability to conceptualize the different parts of the application. Modularity is a worthy goal but, like Microservices, can quickly become hard for most folks to keep track of.

My inclination would be to start with a combined React client (with logical grouping into functional areas) and then look to break it apart when its complexity gets higher (and when we have a clearer picture of what "modularity" looks like).
